LPS-205243.1

Mikellorza opened the pull request.

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

sandrodw3 commented:
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

Mikellorza committed:
LPS-204982 Add friendly url resolver separator company configuration

Mikellorza committed:
LPS-204982 Add friendly url resolver separator manager to manage company configurations

Mikellorza committed:
LPS-204982 Add friendly url separator configuration custom screen

Mikellorza committed:
LPS-204982 Add getKey to identify the friendly url resolver, add isURLSeparatorConfigurable to know if the url separator is configurable or not and add getDefaultURLSeparator to get the default url separator if it is not configured

Mikellorza committed:
LPS-204982 Add action command to save friendly url company configuration

Mikellorza committed:
LPS-204982 Add info for the friendly url separator company configuration view

Mikellorza committed:
LPS-204982 Add language keys

Mikellorza committed:
LPS-204982 Show friendly url separators in the config

Mikellorza committed:
LPS-204982 SF

Mikellorza committed:
LPS-204982 Add FF LPS-203351

sandrodw3 committed:
LPS-204982 Add required files for rendering react and taglibs

sandrodw3 committed:
LPS-204982 Add SeparatorFields component

Mikellorza committed:
LPS-205243 Add method to count LayoutFriendlyURLs by companyId and friendlyURL start with the given friendlyURL or exact match

Mikellorza committed:
LPS-205243 SF

Mikellorza committed:
LPS-205243 buildService

sandrodw3 committed:
LPS-204982 buildLang

Mikellorza committed:
LPS-205243 Add language keys

Mikellorza committed:
LPS-205243 buildLang

Mikellorza committed:
LPS-204982 Add jsp and render SeparatorFields component

Mikellorza committed:
LPS-204982 Types

Mikellorza committed:
LPS-204982 Simplify

Mikellorza committed:
LPS-205243 Delete slash when return the urlSeparator to the view and set it when we save them

Mikellorza committed:
LPS-205243 Add validation to ensure that url separator is not empty, use user values in case if there are errors and add a method to get the errors if there are

Mikellorza committed:
LPS-205243 Normalize friendly url separator to avoid contains characters like %&/()·.{}?¿/ and much more. The separator only contains - numbers(0-9) and letters(a-z lowercase but not in uppercase)

Mikellorza committed:
LPS-205243 Validate maximum and minimum size

Mikellorza committed:
LPS-205243 Validate if there is another urlSeparator with the same value

Mikellorza committed:
LPS-205243 Validate if the separator is a number, since it may conflict with a layout friendlyURL because in last instance we are setting the layoutId as the friendlyURL of the layout.

sandrodw3 committed:
LPS-205243 Send errors to react component

sandrodw3 committed:
LPS-205243 Render general error in jsp

sandrodw3 committed:
LPS-205243 Render errors on each field

sandrodw3 committed:
LPS-205243 Types

Mikellorza committed:
LPS-205243 Validate that the separatorUrl does not conflict with:
1- /-/ especial friendly url separator
2-Other friendly url separator
3-Friendly url mappers
4-Language urls

Mikellorza committed:
LPS-205243 Validate that the separatorUrl does not conflict with:
1-Layout with same friendlyUrl
2-Layout that contains the friendlyUrl

Mikellorza commented:
ci:close
